src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy

```diff
@@ -64,13 +71,86 @@ class NUnit extends ConventionTask {
         project.file(getReportFolder())
     }
 
+    @OutputFile
     File getTestReportPath() {
-        new File(getReportFolderImpl(), reportFileName)
+        // for the non-default nunit tasks, ensure we write the report in a separate file
+        def reportFileNamePrefix = name == 'nunit' ? '' : name
```

Comment:
@sorin-bolos  This is the case for non-default nunit tasks. By default we have a `nunit` task which we're fine to have TestResults.xml. Otherwise we need to prefix it, cause we'll overwrite previous test results (or not even have the tests be executed). This is the case when we create multiple nunit tasks.